### PR TITLE
LayeredArchitecturesTask1

### PR DESCRIPTION
**Pull request by bug2big**
Carting service

### CODE REVIEW

File: net_advanced_course.DAL/Repositories/BaseRepository.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        where TEntity : BaseEntity
+    {
+        protected readonly LiteDatabase _context = null!;
+        protected readonly ILiteCollection<TEntity> _collection;
+
+        public BaseRepository(IOptions<LiteDbSettings> liteDbSettingOptions)
+        {
+            try
+            {
+                var db = new LiteDatabase(liteDbSettingOptions.Value.DatabasePath);
```
Comment (krylatij) [inline net_advanced_course.DAL/Repositories/BaseRepository.cs:18]: Db should be injected via DI in e.g. Startup.cs.

File: net_advanced_course.BLL/Services/Carts/CartingService.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+﻿using net_advanced_course.DAL.Entities;
+using net_advanced_course.DAL.Repositories.Carts;
+
+namespace net_advanced_course.BLL.Services.Carts
+{
+    public class CartingService : ICartingService
+    {
+        private readonly ICartRepository _cartRepository;
```
Comment (krylatij) [inline net_advanced_course.BLL/Services/Carts/CartingService.cs:8]: it should be CartService, not a CartingService

File: net_advanced_course.DAL/Repositories/BaseRepository.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                throw new Exception("Can find or create LiteDb database.", ex);
+            }
+        }
+
+        public IQueryable<TEntity> GetAll()
+        {
+            return _collection.FindAll().AsQueryable();
+        }
+
+        public TEntity GetById(Guid entityId)
```
Comment (krylatij) [inline net_advanced_course.DAL/Repositories/BaseRepository.cs:36]: it is better to use Async versions of methods which are performing i/o operations

**Review by krylatij — CHANGES_REQUESTED** [review]

File: net_advanced_course.DAL/Repositories/BaseRepository.cs (hunk 3 above)
Comment (bug2big) [reply net_advanced_course.DAL/Repositories/BaseRepository.cs]: Fixed

File: net_advanced_course.BLL/Services/Carts/CartingService.cs (hunk 2 above)
Comment (bug2big) [reply net_advanced_course.BLL/Services/Carts/CartingService.cs]: Renamed

File: net_advanced_course.DAL/Repositories/BaseRepository.cs (hunk 1 above)
Comment (bug2big) [reply net_advanced_course.DAL/Repositories/BaseRepository.cs]: Reworked

**Pull request merged**